polar/src/polar.lalrpop

```diff
@@ -202,207 +175,228 @@ RewritableOperator: Operator = {
     "in" => Operator::In,
 };
 
-pub RewrittenOperation: Operation = {
+RewrittenOperation: Value = {
     <op:RewritableOperator> "(" <mut args:(<TermExp> ",")*> <arg:TermExp?> ")" => {
         match arg {
             Some(arg) => args.push(arg),
             None => ()
         };
-        Operation{operator: op, args: args}
+        Value::Expression(Operation{operator: op, args: args})
     },
 };
 
-pub Predicate: Predicate = {
-    <name:Name> "(" <mut args:(<TermExp> ",")*> <arg:TermExp?> ")" => {
-        match arg {
-            Some(arg) => args.push(arg),
-            None => ()
-        };
-        Predicate{name, args}
-    }
-};
 
-Exp11<T>: Term = {
-    <term:Term> if T == "Term" => term,
-    <term:Pattern> if T == "Pattern" => term,
-    "(" <exp:Exp<T>> ")" => exp // seems sorta impossible to support parrens for expressions and list literals.
-}
+New: Value = "new" <literal:Spanned<InstanceLiteralTerm>> => {
+    let args = vec![literal];
+    let op = Operation{operator: Operator::New, args};
+    Value::Expression(op)
+};
 
-Exp10<T>: Term = {
+SpecialExp<T>: Value = {
     // A pattern cannot contain a new operator
-    <new:New> if T == "Term" => new,
-    <start:@L> "cut" => {
+    <New> if T == "Term",
+    "cut" => {
         let args = vec![];
         let op = Operation{operator: Operator::Cut, args};
-        Term::new_from_parser(src_id, start, Value::Expression(op))
+        Value::Expression(op)
     },
-    <start:@L> "forall" "(" <arg1:Exp2<"Term">> "," <arg2:Exp2<"Term">> ")" => {
+    "forall" "(" <arg1:TermExp> "," <arg2:TermExp> ")" => {
```

Comment:
Same question here, what's the implication of changing this from `Exp2`?

Reply:
Only allowing a few more possible combinations. Again, `x and y` now.

```
query> forall(x in [1, 2, 3], x > 0 and x < 4)
Parse error:  did not expect to find the token 'and' at line 1, column 30
query> forall(x in [1, 2, 3], (x > 0 and x < 4))
True
```

Reply:
With this change the former would be valid too.